controller/linodemachine_controller.go

```diff
@@ -117,22 +181,220 @@ func (r *LinodeMachineReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 			LinodeMachine: linodeMachine,
 		},
 	)
+	if err != nil {
+		log.Info("Failed to create machine scope", "error", err.Error())
 
-	return r.reconcile(ctx, machineScope, clusterScope)
+		return ctrl.Result{}, fmt.Errorf("failed to create machine scope: %w", err)
+	}
+
+	return r.reconcile(ctx, machineScope, clusterScope, log)
 }
 
 func (r *LinodeMachineReconciler) reconcile(
 	ctx context.Context,
 	machineScope *scope.MachineScope,
 	clusterScope *scope.ClusterScope,
-) (ctrl.Result, error) {
-	// TODO
-	return ctrl.Result{}, nil
+	logger logr.Logger,
+) (res ctrl.Result, err error) {
+	machineScope.LinodeMachine.Status.Ready = false
+	machineScope.LinodeMachine.Status.FailureReason = nil
+	machineScope.LinodeMachine.Status.FailureMessage = util.Pointer("")
+
+	failureReason := cerrs.MachineStatusError("UnknownError")
+	defer func() {
+		if err != nil {
+			machineScope.LinodeMachine.Status.FailureReason = util.Pointer(failureReason)
+			machineScope.LinodeMachine.Status.FailureMessage = util.Pointer(err.Error())
+
+			conditions.MarkFalse(machineScope.LinodeMachine, clusterv1.ReadyCondition, string(failureReason), clusterv1.ConditionSeverityError, "%s", err.Error())
+
+			r.Recorder.Event(machineScope.LinodeMachine, corev1.EventTypeWarning, string(failureReason), err.Error())
+		}
+
+		if patchErr := machineScope.PatchHelper.Patch(ctx, machineScope.LinodeMachine); patchErr != nil &&
+			client.IgnoreNotFound(patchErr) != nil {
+
+			logger.Error(patchErr, "failed to patch LinodeMachine")
+
+			err = errors.Join(err, patchErr)
+		}
+	}()
+
+	// Delete
```

Comment:
What do you think about breaking the delete/update/create parts of this function out into their own functions? I don't mind if we do it later in another PR (I'm happy to take a crack at it for instance) but I think the function would be more easily understood if we did so.